### ISSUE TITLE
Player does not play on IOS 12 (iPhone 6)

### ISSUE THREAD

**Issue by bahtiyarerden** [DoubleSymmetry/react-native-track-player#1153]
**Describe the bug**
Player does work properly Android and newer IOS versions. But both real device and android i reproduce the issue for IOS 12 (iPhone 6). As you see the log below player can not buffer the song and after that i can use neither skip nor play/pause etc functionality. Player does not respond anything. In the xcode log file `AudioHAL_Client` gives a message `the stream index is out of range`. After that EXCEPTION is thrown.

```
2021-04-09 23:54:58.636217+0300 AppName[56662:23536909] [aqme] 177: timed out after 15.000s (0 1); suspension count=0 (IOSuspensions: )
2021-04-09 23:54:58.637000+0300 AppName[56662:23536909] 239: CA_UISoundClientBase::StartPlaying: AddRunningClient failed (status = -66681).
2021-04-09 23:54:58.656544+0300 AppName[56662:23536910] [javascript] { state: { state: 'playing' } }
2021-04-09 23:54:58.766200+0300 AppName[56662:23537969] [AudioHAL_Client] HALB_IOBufferManager.cpp:226:GetIOBuffer:  HALB_IOBufferManager::GetIOBuffer: the stream index is out of range
2021-04-09 23:54:58.768291+0300 AppName[56662:23537969] [AudioHAL_Client] HALB_IOBufferManager.cpp:226:GetIOBuffer:  HALB_IOBufferManager::GetIOBuffer: the stream index is out of range
2021-04-09 23:54:58.780531+0300 AppName[56662:23537969] [aqme] 254: AQDefaultDevice (1): output stream 0: null buffer
2021-04-09 23:54:58.780714+0300 AppName[56662:23537969] [aqme] 1640: EXCEPTION thrown (-50): -
```

**To Reproduce**
1- Run player iPhone 6 real device or simulator

**Environment (please complete the following information):**
Run `react-native info` in your project and share the content.
```
System:
    OS: macOS 11.1
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Memory: 59.04 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 10.20.1 - /usr/local/opt/node@10/bin/node
    Yarn: 1.17.2 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/opt/node@10/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 21, 22, 23, 24, 28, 29, 30
      Build Tools: 25.0.2, 26.0.1, 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 12.3/12C33 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_162 - /usr/bin/javac
    Python: 2.7.15 - /usr/local/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.4 => 0.63.4
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
What `react-native-track-player` version are you using?
`1.2.7`
Are you testing on a real device or in the simulator? Which OS version are you running?
Both

**Code**
Please, share the code that is causing the issue

**Setup**
```
TrackPlayer.setupPlayer({ waitForBuffer: true });
TrackPlayer.updateOptions({
	stopWithApp: true,
	capabilities: [TrackPlayer.CAPABILITY_PLAY, TrackPlayer.CAPABILITY_PAUSE, TrackPlayer.CAPABILITY_SKIP_TO_NEXT],
	compactCapabilities: [TrackPlayer.CAPABILITY_PLAY, TrackPlayer.CAPABILITY_PAUSE],
});
```
**Listeners**
```
TrackPlayer.addEventListener('playback-state', async state => {
	console.log({ state });
	store.dispatch(SessionActions.updatePlaybackInfo(state));
});

TrackPlayer.addEventListener('playback-track-changed', async track => {
	console.log({ track });
	const { offlineMode } = DeepCopy(store.getState().session);
	if (!offlineMode.isServerDown) {
		if (track.track != null) {
			// Track has changed
			store.dispatch(PlaybackListActions.putTrack(track.track));
		}

		if (track.nextTrack != null) {
			// New track has started playing
			store.dispatch(PlaybackListActions.postTrack(track.nextTrack));
		}
	}
});

TrackPlayer.addEventListener('playback-queue-ended', async track => {
	if (track.track != null) {
		store.dispatch(SessionActions.playerSkip(Reason.QUEUE_ENDED));
	}
});

TrackPlayer.addEventListener('playback-error', async (code, message) => {
	console.log('ERROR OCCURED');
	console.log({ code, message });
	store.dispatch(SessionActions.playerSkip(Reason.PLAYBACK_ERROR));
});

TrackPlayer.addEventListener('remote-next', async () => {
	console.log('remote next');
	store.dispatch(SessionActions.playerSkip(Reason.REMOTE_NEXT));
});

TrackPlayer.addEventListener('remote-pause', async () => {
	console.log('remote pause');
	store.dispatch(SessionActions.playerPause());
});

TrackPlayer.addEventListener('remote-play', async () => {
	console.log('remote play');
	store.dispatch(SessionActions.playerPlay());
});
```
My javascript log is:
<img width="963" alt="image" src="https://user-images.githubusercontent.com/3479123/114237923-73ab2800-998c-11eb-9009-802682196833.png">

After paused player does not respond anything.

**Comment by dmathewwws** [comment]
So, I fixed a similar error when I upgraded to 1.2.7

I was also getting this error in Xcode logs tho. ie) error downloading a remote audio file.

```
Task <2A0B92AD-F845-452D-B67B-E95E04B99A92>.<2> load failed with error Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo
```

So, I made sure 2 things:
1) the link to the audio I wanted to play, works
2) No audio was working on my simulator at all, be it from my app or any other app on the simulator. I even tried going to Youtube + Soundcloud (on Safari) on my simulator and it just wouldn't play any audio from those website.

So, I deleted the simulator and any simulator cache. https://apple.stackexchange.com/questions/306566/can-i-delete-unnecessary-device-simulators-of-xcode

When I downloaded a new simulator, audio (in my app + Youtube on Safari) worked
